x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -217,8 +224,11 @@ protected void masterOperation(
         }
 
         final String rollupIndexName = request.getTargetIndex();
-        // Assert rollup index does not exist
-        MetadataCreateIndexService.validateIndexName(rollupIndexName, state);
+        try {
+            MetadataCreateIndexService.validateIndexName(rollupIndexName, state);
+        } catch (ResourceAlreadyExistsException e) {
+            // ignore index already exists
```

Comment:
If the index exists, I believe we should check the `index.downsample.status` index setting. In case its value is `SUCCESS` we can skip the whole process.

Reply:
Good point, we can check the downsample status and short-circuit the process of attempting to create a new index and start the persistent tasks.